### PR TITLE
Rd 58 endpoint for creating school

### PR DESCRIPTION
**Pull request by NazarShchur**
some school fields changes, add endpoint and tests

### CODE REVIEW

File: src/test/java/com/robosh/ejournal/controller/SchoolControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                .address(ANY_STRING)
+                .name(ANY_STRING)
+                .region(ANY_STRING)
+                .settlementName(ANY_STRING)
+                .settlementType(SettlementType.CITY)
+                .url(ANY_STRING)
+                .build();
+    }
+
+}
```
Comment (SashaRoyenko) [inline src/test/java/com/robosh/ejournal/controller/SchoolControllerTest.java]: New line at the  end

File: src/test/java/com/robosh/ejournal/service/SchoolServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return School.builder()
+                .address(ANY_STRING)
+                .name(ANY_STRING)
+                .region(ANY_STRING)
+                .settlementName(ANY_STRING)
+                .settlementType(SettlementType.CITY)
+                .url(ANY_STRING)
+                .build();
+    }
+}
```
Comment (SashaRoyenko) [inline src/test/java/com/robosh/ejournal/service/SchoolServiceTest.java]: New line at the end

File: src/main/java/com/robosh/ejournal/service/SchoolService.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+@Service
+public class SchoolService {
+
+    private final SchoolRepository schoolRepository;
+
+    private final SchoolMapper schoolMapper;
+
+    public SchoolInfoDto save(SaveSchoolDto saveSchoolDto){
+        School school = schoolMapper.fromSaveSchoolDtoToSchool(saveSchoolDto);
+
```
Comment (SashaRoyenko) [inline src/main/java/com/robosh/ejournal/service/SchoolService.java:24]: Validate?

**Review by SashaRoyenko — CHANGES_REQUESTED** [review]

**Review by StrafeUp — COMMENTED** [review]
School validation

File: src/main/java/com/robosh/ejournal/service/SchoolService.java (hunk 3 above)
Comment (NazarShchur) [reply src/main/java/com/robosh/ejournal/service/SchoolService.java:24]: Is there any custom validation annotation in School?

File: src/test/java/com/robosh/ejournal/service/SchoolServiceTest.java (hunk 2 above)
Comment (NazarShchur) [reply src/test/java/com/robosh/ejournal/service/SchoolServiceTest.java]: There was not new line in other tests i thought it's redundant

File: src/main/java/com/robosh/ejournal/service/SchoolService.java (hunk 3 above)
Comment (SashaRoyenko) [reply src/main/java/com/robosh/ejournal/service/SchoolService.java:24]: Custom no, but you should process not custom

Comment (NazarShchur) [reply src/main/java/com/robosh/ejournal/service/SchoolService.java:24]: How does validation works without it then?

**Review by Ostape — APPROVED** [review]

**Review by StrafeUp — APPROVED** [review]

**Review by SashaRoyenko — APPROVED** [review]

**Review by SashaRoyenko — COMMENTED** [review]
good job

**Review by SashaRoyenko — CHANGES_REQUESTED** [review]
Cool!

**Review by SashaRoyenko — APPROVED** [review]

**Pull request merged**